cpp/scenario/gazebo/src/utils.cpp

```diff
@@ -278,3 +280,103 @@ std::string utils::URDFStringToSDFString(const std::string& urdfString)
 
     return root->Element()->ToString("");
 }
+
+std::vector<double> utils::normalize(const std::vector<double>& input,
+                                     const std::vector<double>& low,
+                                     const std::vector<double>& high)
+{
+    bool okInput = input.size() > 0;
+    bool okLow = low.size() == input.size() || low.size() == 1;
+    bool okHigh = high.size() == input.size() || high.size() == 1;
+
+    if (!okInput || !okLow || !okHigh) {
+        throw std::invalid_argument("Wrong input arguments");
+    }
+
+    std::vector<double> lowBroadcasted = low;
+    std::vector<double> highBroadcasted = high;
+
+    if (low.size() == 1 && input.size() > 1) {
+        lowBroadcasted = std::vector<double>(input.size(), low[0]);
+    }
+
+    if (high.size() == 1 && input.size() > 1) {
+        highBroadcasted = std::vector<double>(input.size(), high[0]);
+    }
+
+    std::vector<double> output;
+    output.resize(input.size());
+
+    auto inputEigen = Eigen::Map<Eigen::ArrayXd>( //
+        const_cast<double*>(input.data()),
+        input.size());
+    auto outputEigen = Eigen::Map<Eigen::ArrayXd>( //
+        output.data(),
+        output.size());
+    auto lowEigen = Eigen::Map<Eigen::ArrayXd>( //
+        lowBroadcasted.data(),
+        lowBroadcasted.size());
+    auto highEigen = Eigen::Map<Eigen::ArrayXd>( //
+        highBroadcasted.data(),
+        highBroadcasted.size());
+
+    if (highEigen.isApprox(lowEigen)) {
+        return input;
+    }
```

Comment:
If high = low, why the input is  unchanged? Should this be  documented?

Reply:
Replied in https://github.com/robotology/gym-ignition/pull/183/files/0c87c0c7d59b105932854219cc32d60f747c0ead#r417522033